integration_test/myxql/storage_test.exs

```diff
@@ -150,7 +150,51 @@ defmodule Ecto.Integration.StorageTest do
     :ok = Ecto.Migrator.up(PoolRepo, num, Migration, log: false)
     {:ok, path} = Ecto.Adapters.MyXQL.structure_dump(tmp_path(), TestRepo.config())
     contents = File.read!(path)
-    assert contents =~ "INSERT INTO `schema_migrations` (version) VALUES ("
+    assert contents =~ "INSERT INTO `ecto_test`.`schema_migrations` (version) VALUES (#{num})"
+  end
+
+  test "dumps structure and schema_migration records from multiple prefixes" do
+    # Create the test_schema schema
+    create_database()
+    prefix = params()[:database]
+
+    # Run migrations
+    version = @base_migration + System.unique_integer([:positive])
+    :ok = Ecto.Migrator.up(PoolRepo, version, Migration, log: false)
+    :ok = Ecto.Migrator.up(PoolRepo, version, Migration, log: false, prefix: prefix)
+
+    config = Keyword.put(TestRepo.config(), :dump_prefixes, ["ecto_test", prefix])
+    {:ok, path} = Ecto.Adapters.MyXQL.structure_dump(tmp_path(), config)
+    contents = File.read!(path)
+
+    assert contents =~ "Current Database: `#{prefix}`"
```

Comment:
Instead of this, are we able to match on the creation of the schema_migration table, like postgres? And same for the other places.

Reply:
I tried that initially but unfortunately, the mysql dump doesn't add any prefix related information there:

```
--
-- Table structure for table `schema_migrations`
--

DROP TABLE IF EXISTS `schema_migrations`;
/*!40101 SET @saved_cs_client     = @@character_set_client */;
/*!50503 SET character_set_client = utf8mb4 */;
CREATE TABLE `schema_migrations` (
  `version` bigint NOT NULL,
  `inserted_at` datetime DEFAULT NULL,
  PRIMARY KEY (`version`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
/*!40101 SET character_set_client = @saved_cs_client */;
```

Reply:
Ah ok thanks for checking. Maybe then we keep what you have about the current database and also match on this?

>CREATE TABLE `schema_migrations`

Just so we know it is capturing the ddl commands.

Reply:
Yep, makes sense. Done!